prefect_kubernetes/pods.py

```diff
@@ -0,0 +1,361 @@
+"""Module for interacting with Kubernetes pods from Prefect flows."""
+
+from typing import Any, Callable, Dict, Optional, Union
+
+from kubernetes.client.exceptions import ApiException
+from kubernetes.client.models import V1DeleteOptions, V1Pod, V1PodList
+from kubernetes.watch import Watch
+from prefect import task
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+
+from prefect_kubernetes.credentials import KubernetesCredentials
+
+
+@task
+async def create_namespaced_pod(
+    kubernetes_credentials: KubernetesCredentials,
+    body: V1Pod,
+    namespace: Optional[str] = "default",
+    **kube_kwargs: Dict[str, Any],
+) -> V1Pod:
+    """Create a Kubernetes pod in a given namespace.
+
+    Args:
+        kubernetes_credentials: `KubernetesCredentials` block for creating
+            authenticated Kubernetes API clients.
+        body: A Kubernetes `V1Pod` specification.
+        namespace: The Kubernetes namespace to create this pod in.
+        **kube_kwargs: Optional extra keyword arguments to pass to the Kubernetes API.
+
+    Returns:
+        A Kubernetes `V1Pod` object.
+
+    Example:
+        Create a pod in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.pods import create_namespaced_pod
+        from kubernetes.client.models import V1Pod
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_pod_metadata = create_namespaced_pod(
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+                body=V1Pod(**{"metadata": {"name": "test-pod"}}),
```

Comment:
Why are you spreading a dictionary to instantiate a `V1Pod`? Would it be cleaner to use keyword arguments instead?

Also, what's the bare minimum that you need to provide to get a pod running? I'd think you need more than just a name.

Reply:
I wanted a way to show that users don't need to import the model from `prefect.client.models` and just pass a dictionary that would be coerced into the corresponding model, but I can switch to kwargs - agreed that's cleaner